### PR TITLE
Fix #7781: TriStateCheckbox add readonly attribute

### LINKED ISSUE #7781: TriStateCheckbox: Add readonly attribute
**Description**
The `<p:triStateCheckbox>` is missing a `readonly` attribute that allows to render the checkbox in a non-disabled fashion, still maintaining it read-only.
It's not even possible to simulate such a behaviour with something like `onclick="return false"` because this component is also missing the `onclick` attribute.

**Describe the solution you would like**
Adding the ability to set `<p:triStateCheckbox readonly="true" />` to make the tri state checkbox look enabled but readonly. This is also useful if you want to submit its value in a form although not letting the user change it.

It seems like this was implemented in primeng: https://github.com/primefaces/primeng/issues/2999

### PR DESCRIPTION
**Pull request by melloware**
I followed what PrimeNG was doing.

### CODE REVIEW

**Comment by melloware** [comment]
Working on Integration tests next.

File: primefaces/src/main/java/org/primefaces/component/tristatecheckbox/TriStateCheckboxRenderer.java [hunk 1]
```diff
@@ -119,11 +120,11 @@ protected void encodeInput(final FacesContext context, final TriStateCheckbox ch
     }
 
     protected void encodeOutput(final FacesContext context, final TriStateCheckbox checkbox, final int valCheck,
-                                final boolean disabled) throws IOException {
+                                final boolean disabled, boolean readonly) throws IOException {
         ResponseWriter writer = context.getResponseWriter();
         String styleClass = createStyleClass(checkbox, null, HTML.CHECKBOX_BOX_CLASS) ;
         styleClass = (valCheck == 1 || valCheck == 2) ? styleClass + " ui-state-active" : styleClass;
-        styleClass = disabled ? styleClass + " ui-state-disabled" : styleClass;
```
Comment (mauromol) [inline primefaces/src/main/java/org/primefaces/component/tristatecheckbox/TriStateCheckboxRenderer.java:126]: I've not tested and I didn't investigate, but are you sure this line should be deleted?

Comment (melloware) [reply primefaces/src/main/java/org/primefaces/component/tristatecheckbox/TriStateCheckboxRenderer.java:126]: No this is a mistake!

**Comment by melloware** [comment]
Added integrations tests for enabled, disabled, and readonly and discovered a bug while writing those tests.

**Pull request merged**